src/Features/Core/Portable/AddFileBanner/AbstractAddFileBannerCodeRefactoringProvider.cs

```diff
@@ -85,21 +88,35 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
                         CodeAction.Create(
                             CodeFixesResources.Add_file_header,
                             _ => AddBannerAsync(document, root, siblingDocument, siblingBanner),
-                            nameof(CodeFixesResources.Add_file_header)),
+                            equivalenceKey: siblingDocument.FilePath),
```

Comment:
> This is changing the equivalence key to something which is not predictable

This should be predictable for the solution snapshot on which refactoring was invoked, but not for test automation as you mention.

> If there is no other way to implement this, we should retain stability of the equivalence key by using the content of the new header instead of the path

I would have preferred to go that route, but the fix all needs to get the file name (we replace file name inside the banner as part of computing the banner for the document to fix). Another couple of alternatives:
1. Thread in an extra property bag (dictionary from string to string, just like on a diagnostic) from original code action to the FixAllAsync API.
2. Thread in the original code action itself to the FixAllAsync API. This simplifies everything, and allows us to share any data from the original code action registration to the FixAllAsync API. However, Cyrus wants to avoid this approach as this may prevent us from running FixAll in a separate process and we wouldn't know how to serialize/deserialize code action across processes.

Reply:
Thanks, addressed.